src/mappings/entities/mapAsset.ts

```diff
@@ -0,0 +1,453 @@
+const excludeTransferManager = (
+  manager: AssetAdvancedCompliance,
+  managers: AssetAdvancedCompliance[]
+) =>
+  managers.filter(
+    e =>
+      e.CountTransferManager !== manager.CountTransferManager ||
+      e.PercentageTransferManager !== manager.PercentageTransferManager
+  );
+// #endregion
+
+// #region ModuleIdEnum.Asset
+const handleCreateAsset = async (params: Record<string, any>, extrinsic: any) => {
+  const { name, ticker, assetType: type, fundingRound, divisible, disableIu, identifiers } = params;
+  const ownerDid = extrinsic.events[1].event.data[0].toString();
+  await Asset.create({
+    id: ticker,
+    ticker,
+    name,
+    type,
+    fundingRound,
+    isDivisible: divisible,
+    isFrozen: false,
+    isUniquenessRequired: !disableIu,
+    documents: [],
+    identifiers: formatAssetIdentifiers(identifiers),
+    ownerDid,
+    fullAgents: [ownerDid],
+    holders: [],
+    totalSupply: '0',
+    totalTransfers: '0',
+    compliance: { isPaused: false, sender: [], receiver: [], advanced: [] },
+  }).save();
+};
+
+const handleRenameAsset = async (params: Record<string, any>) => {
+  const { ticker, name: newName } = params;
+  const asset = await getAsset(ticker);
+  asset.name = newName;
+  await asset.save();
+};
+
+const handleSetFundingRound = async (params: Record<string, any>) => {
+  const { ticker, name: newFundingRound } = params;
+  const asset = await getAsset(ticker);
+  asset.fundingRound = newFundingRound;
+  await asset.save();
+};
+
+type AssetNewDocument = {
+  name: string;
+  uri: string;
+  content_hash?: any;
+  doc_type?: string;
+  filing_date?: Date;
+};
+
+const handleAddDocuments = async (params: Record<string, any>, extrinsic: any) => {
+  const { ticker, docs } = params;
+  const asset = await getAsset(ticker);
+  asset.documents = docs.map((doc: AssetNewDocument, i: number) => ({
+    id: Number(extrinsic.events[i].event.data[2].toString()),
+    name: doc.name,
+    link: doc.uri,
+  }));
+  await asset.save();
+};
+
+const handleRemoveDocuments = async (params: Record<string, any>) => {
+  const { ticker, ids } = params;
+  const asset = await getAsset(ticker);
+  asset.documents = asset.documents.filter(doc => !ids.includes(doc.id));
+  await asset.save();
+};
+
+const handleUpdateIdentifiers = async (params: Record<string, any>) => {
+  const { ticker, identifiers: newIdentifiers } = params;
+  const asset = await getAsset(ticker);
+  asset.identifiers = formatAssetIdentifiers(newIdentifiers);
+  await asset.save();
+};
+
+const handleMakeDivisible = async (params: Record<string, any>) => {
+  const { ticker } = params;
+  const asset = await getAsset(ticker);
+  asset.isDivisible = true;
+  await asset.save();
+};
+
+const handleIssue = async (params: Record<string, any>) => {
+  const { ticker, amount } = params;
+  const asset = await getAsset(ticker);
+  const formattedAmount = chainAmountToBigNumber(amount);
+  const ownerAmount = getHolderAmount(asset.ownerDid, asset.holders);
```

Comment:
I don't think the owner is always the one who will receive the issuance: https://polymathnetwork.slack.com/archives/C01RLA867C5/p1644335531529869?thread_ts=1644331664.658489&cid=C01RLA867C5

Reply:
You're right, technically it's the PIA, which is now external/full agent, no?
I'll work on it but this should be fine for now since owner/PIA are currently the same, and we're transferring full agency with ownership.

Reply:
If the constraint is that the PIA is always the owner then this is OK, but I thought since we have the term PIA that implies they might not always be the same.

Reply:
Let's revisit this later. With the new projects kicking off we'll need this new asset entity.